include/aws/sdkutils/endpoints_rule_engine.h

```diff
@@ -164,7 +166,8 @@ AWS_SDKUTILS_API const struct aws_string *aws_endpoints_ruleset_get_service_id(
  */
 AWS_SDKUTILS_API struct aws_endpoints_rule_engine *aws_endpoints_rule_engine_new(
     struct aws_allocator *allocator,
-    struct aws_endpoints_ruleset *ruleset);
+    struct aws_endpoints_ruleset *ruleset, 
+    struct aws_partitions_config *partitions_config);
```

Comment:
trivial/debatable: is it likely that more things will factor into making a new `aws_endpoints_rule_engine` in the future? is it worth having an options-struct instead of individual arguments?

if you think it's unlikely, we can leave this as-is, since it seems like options-structs themselves could need a revamp in the near-future

Reply:
on const partitions_config i think its worth more discussion. the only reason its not const is because of ref counting. 

Re: wrapping args in options. It's really unlikely that resolution itself will need more arguments in the near future. We did discuss potential needs for caching, but current alignment is that it should happen outside of rule engine, either on client side or crt will introduce a wrapper on top of engine that will provide caching